src/directives/ng-file-select.ts

```diff
@@ -25,8 +25,23 @@ export class NgFileSelect {
 
   onFiles(): void {
     let files = this.el.nativeElement.files;
-    if (files.length) {
+    let invalidExtentionFiles: Array<string> = [];
+
+    if (files.length && this.uploader.allowedExtensions.length) {
+      for (var i = 0; i < files.length; i++) {
```

Comment:
Okay then, thanks for your PR.
I will test it and merge if everything works. Can you rebase please, I
updated master couple of hours ago. Thanks.

On Thursday, 18 August 2016, Popat <notifications@github.com
<javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:

> In src/directives/ng-file-select.ts
> <https://github.com/jkuri/ng2-uploader/pull/62#discussion_r75282710>:
>
> > @@ -25,8 +25,23 @@ export class NgFileSelect {
> >
> >    onFiles(): void {
> >      let files = this.el.nativeElement.files;
> > -    if (files.length) {
> > +    let invalidExtentionFiles: Array<string> = [];
> > +
> > +    if (files.length && this.uploader.allowedExtensions.length) {
> > +      for (var i = 0; i < files.length; i++) {
>
> files is a FileList object, If we use forEach then it will through
> following error.
>
> browser_adapter.ts:74 ORIGINAL EXCEPTION: TypeError: files.forEach is not
> a function
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jkuri/ng2-uploader/pull/62/files/4af1135ce1bda357d709670c6bf1b0a124284573#r75282710>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABtntnoV41s7iIKb2QvPiK7QuAEQLXDLks5qhDEKgaJpZM4JmUc6>
> .
>